File: uiweb/src/panel/MiniCBottomPanelModel.ts
import type { JavaMirrorFile } from "../translation/javaMirror";

export const miniCBottomPanelModelMirror = {
  "javaPath": "src/main/java/minic/uilocal/panel/MiniCBottomPanelModel.java",
  "webPath": "uiweb/src/panel/MiniCBottomPanelModel.ts",
  "packageName": "minic.uilocal",
  "exportName": "MiniCBottomPanelModel",
  "kind": "record",
  "imports": [
    "java.util.List",
    "java.util.Objects"
  ],
  "fields": [],
  "methods": []
} as const satisfies JavaMirrorFile;

export class MiniCBottomPanelModel {
  static readonly mirror = miniCBottomPanelModelMirror;

  readonly mirror = miniCBottomPanelModelMirror;

  readonly problems: readonly string[];

  readonly output: readonly string[];

  readonly terminal: readonly string[];

  constructor(
    problems: readonly string[],
    output: readonly string[],
    terminal: readonly string[],
  ) {
    this.problems = [...problems];
    this.output = [...output];
    this.terminal = [...terminal];
  }

  summary(): string {
    return `MiniCBottomPanelModel: ${this.mirror.methods.length} methods, ${this.mirror.fields.length} fields`;
  }
}

export default MiniCBottomPanelModel;

File: src/main/java/minic/uilocal/panel/MiniCBottomPanelModel.java
package minic.uilocal;

import java.util.List;
import java.util.Objects;

/**
 * 底部面板展示数据。
 *
 * @param problems 问题列表
 * @param output 输出列表
 * @param terminal 终端日志列表
 */
public record MiniCBottomPanelModel(
        List<String> problems,
        List<String> output,
        List<String> terminal
) {
    public MiniCBottomPanelModel {
        Objects.requireNonNull(problems, "problems");
        Objects.requireNonNull(output, "output");
        Objects.requireNonNull(terminal, "terminal");
        problems = List.copyOf(problems);
        output = List.copyOf(output);
        terminal = List.copyOf(terminal);
    }
}
